GitHub.Authentication/Authentication.cs

```diff
@@ -336,5 +341,15 @@ public override void SetCredentials(TargetUri targetUri, Credential credentials)
         /// </param>
         /// <param name="result">The result of the interactive authentication attempt.</param>
         public delegate void AuthenticationResultDelegate(TargetUri targetUri, GitHubAuthenticationResultType result);
+
+        static Uri NormalizeUri(Uri targetUri)
+        {
+            // Special case for gist.github.com which are git backed repositories under the hood.
+            // Credentials for these repos are the same as the one stored with "github.com"
+            if (targetUri.DnsSafeHost.StartsWith("gist.", StringComparison.OrdinalIgnoreCase) && targetUri.DnsSafeHost.EndsWith(".github.com", StringComparison.OrdinalIgnoreCase))
```

Comment:
Are we better off just always returning github.com, or do you have specialized sub-domains under the github.com root?

Reply:
I wanted to play it safe just in case we ever do have specialized sub-domains. In practice, nobody should be passing in `foo.github.com` and if they do, sorry!